### PR TITLE
hh-suite: init at 3.3.0

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

The HH-suite is an open-source software package for sensitive protein sequence searching based on the pairwise alignment of hidden Markov models (HMMs).

repo
https://github.com/soedinglab/hh-suite

paper
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3019-7

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — APPROVED** [review]
LGTM

File: pkgs/applications/science/biology/hh-suite/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  nativeBuildInputs = [ cmake xxd ];
+  cmakeFlags =
+    [ "-DHAVE_SSE2=1" ]
+    ++ lib.optional enableAvx2 "-DHAVE_AVX2=1"
+    ++ lib.optional enableSse4_1 "-DHAVE_SSE4_1=1";
+
+  buildInputs =
+    lib.optional stdenv.cc.isClang openmp
+    ++ lib.optional enableMpi mpi;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/hh-suite/default.nix:31]:
```suggestion

  cmakeFlags = [ "-DHAVE_SSE2=1" ]
    ++ lib.optional stdenv.hostPlatform.avx2Support "-DHAVE_AVX2=1"
    ++ lib.optional stdenv.hostPlatform.sse4_1Support "-DHAVE_SSE4_1=1";

  buildInputs = lib.optional stdenv.cc.isClang openmp
    ++ lib.optional enableMpi mpi;
```

File: pkgs/applications/science/biology/hh-suite/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, xxd
+, enableAvx2 ? stdenv.hostPlatform.avx2Support
+, enableSse4_1 ? stdenv.hostPlatform.sse4_1Support
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/hh-suite/default.nix:7]:
```suggestion
```

File: pkgs/applications/science/biology/hh-suite/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "soedinglab";
+    repo = "hh-suite";
+    rev = "v${version}";
+    hash = "sha256-kjNqJddioCZoh/cZL3YNplweIGopWIGzCYQOnKDqZmw=";
+  };
+
+  nativeBuildInputs = [ cmake xxd ];
+  cmakeFlags = [ "-DHAVE_SSE2=1" ]
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/hh-suite/default.nix:22]:
```suggestion
  cmakeFlags = lib.optional stdenv.hostPlatform.isx86 "-DHAVE_SSE2=1"
```
aarch64 does not support that I just remembered

Comment (natsukium) [reply pkgs/applications/science/biology/hh-suite/default.nix]: Thanks!
Also, I added the flag for ARM.

**Pull request merged**